Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift

```diff
@@ -14,7 +14,7 @@ struct SponsoredTileDataUtility: SponsoredTileDataUtilityInterface {
         guard let defaultSearchEngine = searchEngine,
               let secondLevelDomain = site.secondLevelDomain else { return true }
 
-        let hasMatchedEngine = defaultSearchEngine.shortName.caseInsensitiveCompare(secondLevelDomain) == .orderedSame
+        let hasMatchedEngine = defaultSearchEngine.shortName.localizedCaseInsensitiveContains(secondLevelDomain)
```

Comment:
Can we add a test for this, so we cover that we do the right thing when we need that `contains`?

Reply:
It looks like `testShouldAdd_filterSite_whenCaseInsentiveIsUsed` is already covering the case when the default search engine's short name matches the SLD of a site. 

Is there something additional you're looking for?

Reply:
Oh that's for the case insensitive (so `TeSt` website will be filtered even when the search engine is `test`). The extra test case I am mentioning is to have a `test.com` website filtered when the search engine is `test`. So we make sure we don't check for equality, but with contains.